File: MotivApp/app/src/main/java/com/motiv/piotr/dao/MyRoomDatabase.java
package com.motiv.piotr.dao;

import androidx.fragment.app.*;
import androidx.room.*;
import com.motiv.piotr.Link;
import com.motiv.piotr.Links;
import com.motiv.piotr.Meta;
import com.motiv.piotr.Photo;
import com.motiv.piotr.PhotosListResponse;
import com.motiv.piotr.Post;
import com.motiv.piotr.PostsListResponse;
import com.motiv.piotr.RateLimit;
import com.motiv.piotr.User;
import com.motiv.piotr.UserResponse;
import com.motiv.piotr.UsersListResponse;
import dagger.*;
import dagger.android.*;
import dagger.android.support.*;
import javax.inject.*;

@TypeConverters({MyTypeConverter.class})
@Database(
    entities = {
        Meta.class,
        Photo.class,
        User.class,
        UserResponse.class,
        PhotosListResponse.class,
        Links.class,
        RateLimit.class,
        Post.class,
        PostsListResponse.class,
        Link.class,
        UsersListResponse.class
    },
    version = 1
)
public abstract class MyRoomDatabase extends RoomDatabase {

    public static final String DATABASE_NAME = "test";

    public abstract com.motiv.piotr.dao.UserResponseDao userResponseDao();

    public abstract com.motiv.piotr.dao.MetaDao metaDao();

    public abstract com.motiv.piotr.dao.RateLimitDao rateLimitDao();

    public abstract com.motiv.piotr.dao.UserDao userDao();

    public abstract com.motiv.piotr.dao.LinksDao linksDao();

    public abstract com.motiv.piotr.dao.LinkDao linkDao();

    public abstract com.motiv.piotr.dao.UsersListResponseDao usersListResponseDao();

    public abstract com.motiv.piotr.dao.PostsListResponseDao postsListResponseDao();

    public abstract com.motiv.piotr.dao.PostDao postDao();

    public abstract com.motiv.piotr.dao.PhotosListResponseDao photosListResponseDao();

    public abstract com.motiv.piotr.dao.PhotoDao photoDao();
}

File: MotivApp/app/src/main/java/com/motiv/piotr/dao/MyRoomDatabase.kt
 
package com.motiv.piotr.dao
import androidx.room.*
import com.motiv.piotr.Link
import com.motiv.piotr.Links
import com.motiv.piotr.Meta
import com.motiv.piotr.Photo
import com.motiv.piotr.PhotosListResponse
import com.motiv.piotr.Post
import com.motiv.piotr.PostsListResponse
import com.motiv.piotr.RateLimit
import com.motiv.piotr.User
import com.motiv.piotr.UserResponse
import com.motiv.piotr.UsersListResponse
@TypeConverters(MyTypeConverter::class)
@Database(entities = [PostsListResponse::class, PhotosListResponse::class, UsersListResponse::class, RateLimit::class, Links::class, User::class, Post::class, UserResponse::class, Photo::class, Link::class, Meta::class], version = 1)
public abstract class MyRoomDatabase : RoomDatabase() {

    companion object {
        val DATABASE_NAME: String = "test"
    }

    public abstract fun userResponseDao(): com.motiv.piotr.dao.UserResponseDao
    public abstract fun metaDao(): com.motiv.piotr.dao.MetaDao
    public abstract fun rateLimitDao(): com.motiv.piotr.dao.RateLimitDao
    public abstract fun userDao(): com.motiv.piotr.dao.UserDao
    public abstract fun linksDao(): com.motiv.piotr.dao.LinksDao
    public abstract fun linkDao(): com.motiv.piotr.dao.LinkDao
    public abstract fun usersListResponseDao(): com.motiv.piotr.dao.UsersListResponseDao
    public abstract fun postsListResponseDao(): com.motiv.piotr.dao.PostsListResponseDao
    public abstract fun postDao(): com.motiv.piotr.dao.PostDao
    public abstract fun photosListResponseDao(): com.motiv.piotr.dao.PhotosListResponseDao
    public abstract fun photoDao(): com.motiv.piotr.dao.PhotoDao
}
